extensions/sockets/src/main/java/org/mule/module/socket/api/provider/tcp/TcpListenerProvider.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.api.provider.tcp;
+
+import org.mule.module.socket.api.config.ListenerConfig;
+import org.mule.module.socket.api.connection.tcp.TcpListenerConnection;
+import org.mule.module.socket.api.protocol.SafeProtocol;
+import org.mule.module.socket.api.protocol.TcpProtocol;
+import org.mule.module.socket.api.socket.tcp.TcpServerSocketProperties;
+import org.mule.module.socket.api.connection.ConnectionSettings;
+import org.mule.module.socket.api.socket.tcp.DefaultTcpServerSocketProperties;
+import org.mule.module.socket.internal.SocketUtils;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.connection.ConnectionHandlingStrategy;
+import org.mule.runtime.api.connection.ConnectionHandlingStrategyFactory;
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.connection.ConnectionValidationResult;
+import org.mule.runtime.extension.api.annotation.Alias;
+import org.mule.runtime.extension.api.annotation.Parameter;
+import org.mule.runtime.extension.api.annotation.ParameterGroup;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+
+/**
+ * A {@link ConnectionProvider} which provides instances of
+ * {@link TcpListenerConnection}
+ *
+ * @since 4.0
+ */
+@Alias("tcp-listener")
+public final class TcpListenerProvider implements ConnectionProvider<ListenerConfig, TcpListenerConnection>
+{
+
+    /**
+     * Parameters that indicates where the TCP server socket should listen for new connections
```

Comment:
these parameters don't indicate that it should listen. Listening is its nature. They configure the listener itself.